metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -0,0 +1,147 @@
+package scala.meta.internal.metals
+
+import java.nio.file.Path
+import java.util
+import java.util.Collections
+import java.util.Optional
+
+import scala.collection.mutable
+import scala.util.Try
+
+import scala.meta._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.MessageType
+import org.eclipse.{lsp4j => l}
+import scalafix.interfaces.Scalafix
+import scalafix.interfaces.ScalafixArguments
+
+case class ScalafixProvider(
+    buildTargets: BuildTargets,
+    buffers: Buffers,
+    workspace: AbsolutePath,
+    embedded: Embedded,
+    statusBar: StatusBar,
+    icons: Icons,
+    languageClient: MetalsLanguageClient
+) {
+  import ScalafixProvider._
+  private val scalafixCache = mutable.Map.empty[String, Scalafix]
+
+  def organizeImports(file: AbsolutePath): List[l.TextEdit] = {
+    val fileInput = file.toInput
+    val unsavedFile = file.toInputFromBuffers(buffers)
+    if (fileInput.text.diff(unsavedFile.text).nonEmpty) {
+      scribe.info(s"Organize imports requires saving the file first")
+      languageClient.showMessage(
+        MessageType.Warning,
+        s"Save ${file.toNIO.getFileName} before organizing imports"
+      )
+      Nil
+    } else {
+      val baseMatcher = scalafix.internal.v1.Args.baseMatcher
+      if (baseMatcher.matches(file.toNIO)) {
+        val scalafixConfPath = workspace.resolve(scalafixFileName)
+        val scalafixConf: Optional[Path] =
+          if (scalafixConfPath.isFile) Optional.of(scalafixConfPath.toNIO)
+          else Optional.empty()
```

Comment:
Nit: in other places we generally use the Scala type and then convert to Java types on demand.

```suggestion
        val scalafixConf =
          if (scalafixConfPath.isFile) Some(scalafixConfPath.toNIO)
          else None
```

and then `scalafixConf.asJava` where needed